### ISSUE TITLE
[Feature] Optimization required: It is taking too long for the grid to show

### ISSUE THREAD

**Issue by chriscdn** [fluttercandies/flutter_photo_manager#773]
**Is your feature request related to a problem? Please describe.**

This is a continuation of the performance issue [reported here](https://github.com/fluttercandies/flutter_wechat_assets_picker/issues/116). The previous issue was closed since it could not be replicated. I suspect it has to do with the device state (i.e., device model, number of photos on the device, iCloud, etc.).

I investigated the issue on my device, and the purpose of this issue is to summarize my findings.  I hope it helps resolve the problem.

The grid can take a long time to show before the user can start selecting photos.  In my case, it can take anywhere from 5-10s. One of my users claims it takes **six minutes** to show on their device, but I have no further information.

I did some tracing and suspect it has to do, partially, with [PMManager.m](https://github.com/fluttercandies/flutter_photo_manager/blob/ca2187604e9a85a18889c6d23f740963553447b7/ios/Classes/core/PMManager.m) on line 170 (called from `getGalleryList`):

```
PHFetchResult<PHAsset *> *fetchResult = [PHAsset fetchAssetsInAssetCollection:assetCollection options:options];
```

This is inside a loop, and is called on each album and smart album in the device.  I timed the method, and it takes ~2s alone for the iOS "Recents" album, and ~1.2s for the "Bursts" album (I don't even have many photos in the Bursts album).  In all, I have 72 albums on my device, and the sum of these timings is ~6s.

I also timed the `getGalleryList` method, and it was ~7s.  It implies much of the time is waiting on these `fetchAssetsInAssetCollection` calls.

To complicate matters, everything is faster after the device is rebooted.  So something is happening to the state of the device, which slows down these calls.  After time (a few days after rebooting?) it's slow again.  I can't tell you what causes this.

**Describe the solution you'd like**

It seems like a problem on Apple's side.  Clearly, we can't ask users to reboot their device each time this happens, so my immediate question is whether the `fetchAssetsInAssetCollection` calls can be reduced or refactored somehow?  Perhaps some caching, with a method to pre cache the values before the user opens the picker?  Perhaps caching can be expired with [an observer](https://developer.apple.com/documentation/photokit/phphotolibrary/observing_changes_in_the_photo_library?language=objc)?

**Describe alternatives you've considered**

I've tried almost every photo picker on pub.dev, and each has their limitations.  The `flutter_wechat_assets_picker` is the only one that covers my use case (it's great!), with the only issue being performance.

**Additional context**

My testing was with:

- iPhone SE gen 2, iOS 15.4.1
- Photos are stored in iCloud
- ~22'000 photos, 72 albums
- called `PhotoManager.clearFileCache()` between tests

My call to `pickAssets` is nothing special:

```
final List<AssetEntity> images = await AssetPicker.pickAssets(
  context,
  pickerConfig: AssetPickerConfig(
    maxAssets: maxAssets,
    requestType: RequestType.image,
  ),
)
```

I suspect there are also performance issues with thumbnail loading once the device gets into this "slow" state.  I'll open a separate issue if I find more information.

**Comment by AlexV525** [comment]
Duplicate with #750. You can update more info if you found the tracking issue is missing them.
